internal/dkg/dkg_process.go

```diff
@@ -116,6 +116,9 @@ func (d *Process) Close() {
 	if err != nil {
 		d.log.Errorw("error closing the database", "err", err)
 	}
+	// this sleep ensures that any in progress DKG will register the closure of the process
+	// before trying to write on the `completedDKGs` channel
+	time.Sleep(1 * time.Second)
```

Comment:
This is horrible T.T

Reply:
have just closed the channel on `<- p.done` now in the other PR so we can remove this

Reply:
I'll rebase shortly. Somehow I didn't see some of these comments from monday so I'm working through them before rebasing